src/controllers/users.contoller.js

```diff
@@ -0,0 +1,75 @@
+const { asyncHandler } = require('../utils/asyncHandler');
+const { usersServices } = require('../services/users.services');
+const { userDto } = require('../dto/user.dto');
+const { ApiError } = require('../exeptions/auth.error');
+const { checkUserPassword } = require('../utils/users/checkUserPassword');
+const { nameSchema } = require('../utils/validation/name');
+const { sendEmailNotify } = require('../services/email.services');
+const { emailSchema } = require('../utils/validation/email');
+
+const getUsers = async (_, res) => {
+  const users = await usersServices.getUsers();
+
+  const usersDto = users.map((user) => userDto(user));
+
+  res.status(200).json(usersDto);
+};
+
+const updateUser = async (req, res) => {
+  const { id } = req.params;
+
+  const user = await usersServices.getUsersById(id);
+
+  if (!user) {
+    throw ApiError.badRequest("The user doesn't exist");
+  }
+
+  const { name, email, newPassword, confirmedPassword, oldPassword } = req.body;
+
+  if (oldPassword) {
+    await checkUserPassword(
+      oldPassword,
+      user.password,
+      newPassword,
+      confirmedPassword,
+    );
+  }
```

Comment:
The `checkUserPassword` function is called only if `oldPassword` is provided. Ensure that this function is necessary in this context, as it might not be needed if `oldPassword` is not required for the update.